src/libraries/System.Net.Sockets/src/System/Net/Sockets/SafeSocketHandle.Unix.cs

```diff
@@ -14,17 +14,88 @@ public partial class SafeSocketHandle
     {
         private int _receiveTimeout = -1;
         private int _sendTimeout = -1;
-        private bool _nonBlocking;
         private SocketAsyncContext? _asyncContext;
 
-        internal bool LastConnectFailed { get; set; }
-        internal bool DualMode { get; set; }
-        internal bool ExposedHandleOrUntrackedConfiguration { get; private set; }
-        internal bool PreferInlineCompletions { get; set; } = SocketAsyncEngine.InlineSocketCompletionsEnabled;
-        internal bool IsSocket { get; set; } = true; // (ab)use Socket class for performing async I/O on non-socket fds.
+        [Flags]
+        private enum Flags : byte
+        {
+            NonBlocking = 1,
+            LastConnectFailed = 2,
+            DualMode = 4,
+            ExposedHandleOrUntrackedConfiguration = 8,
+            PreferInlineCompletions = 16,
+            IsSocket = 32,
+            IsDisconnected = 64,
 #if SYSTEM_NET_SOCKETS_APPLE_PLATFROM
-        internal bool TfoEnabled { get; set; }
+            TfoEnabled = 128
 #endif
+        }
+
+        private Flags _flags = Flags.IsSocket | (SocketAsyncEngine.InlineSocketCompletionsEnabled ? Flags.PreferInlineCompletions : 0);
+
+        internal bool LastConnectFailed
+        {
+            get => (_flags & Flags.LastConnectFailed) != 0;
+            set
+            {
+                if (value) _flags |= Flags.LastConnectFailed;
+                else _flags &= ~Flags.LastConnectFailed;
```

Comment:
nit: These are reusing same pattern, perhaps you can write a simple helper for this purpose.